index.js

```diff
@@ -131,17 +136,30 @@ AudioManager.prototype.setVolume = function (channelId, volume, muted) {
 		// a sound was playing, channel becomes muted
 		channel.loopSound.stop();
 		// TODO: unload sound ?
-	} else if (channel.loopSound) {
-		// a sound is loaded in channel, updating volume & playback
+	} else if (channel.loopSound && channel.loopSound.id === channel.loopId) {
+		// correct sound is loaded in channel, updating volume & playback
 		channel.loopSound.setVolume(Math.max(0, Math.min(1, volume * channel.loopVol)));
 		if (wasChannelMuted) { channel.loopSound.play(); }
 	} else if (!channel.muted) {
-		// no sounds are loaded in channel, channel is unmutted
+		// sound is not loaded in channel, channel has been unmutted
 		this.playLoopSound(channelId, channel.loopId, channel.loopVol);
 	}
 };
 
 //▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
+/** Mute / unmute all channels
+ *
+ * @param {boolean} [muted] - Should all channels be muted. If not specified, function will behave as toggle
+ */
+AudioManager.prototype.muteAll = function (muted) {
```

Comment:
The naming is a bit confusing (since we're passing a boolean). The name `muteAll` implies `true`. I don't have spectacular alternatives, but I'll try anyway:

- mute + unmute
- setMute(bool)

I think the `All` word is redundant, since you're talking to the audio manager, not a single channel.